src/Utilities/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataAnalysis.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.PointsToAnalysis;
+
+namespace Analyzer.Utilities.FlowAnalysis.Analysis.TaintedDataAnalysis
+{
+    using PointsToAnalysisResult = DataFlowAnalysisResult<PointsToBlockAnalysisResult, PointsToAbstractValue>;
+
+    internal partial class TaintedDataAnalysis : ForwardDataFlowAnalysis<TaintedDataAnalysisData, TaintedDataAnalysisContext, TaintedDataAnalysisResult, TaintedDataBlockAnalysisResult, TaintedDataAbstractValue>
+    {
+        private static readonly TaintedDataAnalysisDomain TaintedDataAnalysisDomainInstance = new TaintedDataAnalysisDomain(CoreTaintedDataAnalysisDataDomain.Instance);
+
+        private TaintedDataAnalysis(TaintedDataOperationVisitor operationVisitor)
+            : base(TaintedDataAnalysisDomainInstance, operationVisitor)
+        {
+        }
+
+        internal static TaintedDataAnalysisResult GetOrComputeResult(
+            ControlFlowGraph cfg,
+            Compilation compilation,
+            ISymbol containingMethod,
+            TaintedDataSymbolMap<SourceInfo> taintedSourceInfos,
+            TaintedDataSymbolMap<SanitizerInfo> taintedSanitizerInfos,
+            TaintedDataSymbolMap<SinkInfo> taintedSinkInfos)
+        {
+            WellKnownTypeProvider wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilation);
+            PointsToAnalysisResult pointsToAnalysisResult = PointsToAnalysis.GetOrComputeResult(
```

Comment:
>PointsToAnalysis.GetOrComputeResult [](start = 60, length = 35)

Just thinking out loud... For CA2100, we also use the [ValueContentAnalysis results](https://github.com/dotnet/roslyn-analyzers/blob/features/dataflow/src/Microsoft.NetCore.Analyzers/Core/Data/ReviewSqlQueriesForSecurityVulnerabilities.cs#L218) - this helps when we know for sure that a string is a constant at a given point using predicate analysis. I am not sure if that would be useful for your case, given your analysis is not specific to strings, while CA2100 is specific to strings.